### PR TITLE
AO-10251: Service name cleansing

### PR DESCRIPTION
**Pull request by jiwen624**
Jira ticket: 
- https://swicloud.atlassian.net/browse/AO-10242
- https://swicloud.atlassian.net/browse/AO-10251

service name cleansing:
- convert all characters into lowercase
- replace spaces with hyphen
- remove invalid characters

### CODE REVIEW

File: v1/ao/internal/config/validators.go [hunk 1]
```diff
@@ -26,17 +28,54 @@ func NonDefaultEnv(env string, val string) string {
...
+//
+// It doesn't touch the service key but does the following to the original
+// service name:
+// - convert all characters to lowercase
+// - convert spaces to hyphens
+// - remove invalid characters ( [^a-z0-9.:_-])
 func ToServiceKey(s string) interface{} {
-	return s
+	keyS := strings.Split(s, serviceKeyDelimiter)
+	if len(keyS) != serviceKeyPartsCnt {
```
Comment (patsonluk) [inline v1/ao/internal/config/validators.go]: What if the service name itself contains `:`? Think that's a valid character and this logic might mistakenly thinks that's not a valid service key? ie len(keyS) would be > 2

Comment (jiwen624) [reply v1/ao/internal/config/validators.go]: Hmm it makes sense, thanks.

**Comment by jiwen624** [comment]
@patsonluk please see the commit above, thanks.

**Comment by jiwen624** [comment]
The commit above adds the log printing of the converted service key, and also indicates the service key acceptance policy.

**Pull request merged**